lib/node/fullnode.js

```diff
@@ -646,6 +646,80 @@ class FullNode extends Node {
 
     return this.filterindex.getFilter(hash);
   }
+
+  /**
+   * Retrieve compact filter header by hash.
+   * @param {Hash | Number} hash
+   * @returns {Promise} - Returns {@link hash}.
+   */
+
+  async getBlockFilterHeader(hash) {
+    if (!this.filterindex)
+      return null;
+
+    if (typeof hash === 'number')
+      hash = await this.chain.getHash(hash);
+
+    if (!hash)
+      return null;
+
+    return this.filterindex.getFilterHeader(hash);
+  }
+
+  /**
+   * Retrieve compact filter headers by filterType, startHeight and stopHeight.
+   * @param {Number} filterType
+   * @param {Number} startHeight
+   * @param {Hash} stopHash
+   * @returns {Promise} - Returns {@link Object(
+   *   filterType: Number,
+   *   stopHash: Hash,
+   *   previousFilterHeader: Buffer,
+   *   filterHeaderHashes: hash[]
+   * )}.
+   */
+
+  async getBlockFilterHeaders(filterType, startHeight, stopHash) {
+    if (!stopHash)
+      return null;
+
+    const stopHeight = await this.chain.getHeight(stopHash);
+
+    if (!stopHeight)
+      return null;
+
+    if (typeof startHeight === 'number')
+      if (startHeight > stopHeight || stopHeight - startHeight >= 2000)
+        return null;
+
+    const headerHashes = [];
+
+    const firstBlockHash = await this.chain.getHash(startHeight);
+    const firstBlockHeader = await this.getBlockFilterHeader(firstBlockHash);
+
+    if (firstBlockHeader == null)
+      return null;
+
+    const prevBlockHash = firstBlockHeader.prevBlock;
+    const prevBlockHeader = await this.getBlockFilterHeader(prevBlockHash);
+
+    for (let i = startHeight; i <= stopHeight; i++) {
+      const hash = await this.chain.getHash(i);
```

Comment:
I think using `Promise.all()` here instead of `await` would make things faster.

Reply:
Yeah, that's a great suggestion will do that!